rdt-app/src/main/java/io/ona/rdt_app/viewholder/PatientRegisterViewHolder.java

```diff
@@ -112,7 +113,12 @@ public RegisterViewHolder createViewHolder(ViewGroup parent) {
             @Override
             public void onClick(View v) {
                 try {
-                    new RDTJsonFormUtils().launchForm((Activity) context, RDT_TEST_FORM, view.getTag(R.id.base_entity_id).toString());
+                    String patientNameAndAge = ((TextView) view.findViewById(R.id.tv_patient_name_and_age)).getText().toString();
+                    String patientName = patientNameAndAge.split(",")[0];
```

Comment:
Consider saving an instance of the view holder on `getView` method so that there is no view traversal again 
 e.g `view.findViewById`

Reply:
Wouldn't this always have reference to the last instantiated row?

This wouldn't be desirable.

Reply:
Ideally the click listener should not be when creating views but when rendering the views in recyler views. In this case click handlers should be attached on onBind which calls getView.

I think it would be better to 
1. Proceed with current implementation and Tag the patient object in the createViewHolder holder , then retrieve the object using the tag when inside on onClick
2. Add onclick on getView when you have a patient reference as already done on attachPatientOnclickListener